xcode/Subconscious/Shared/Library/URLUtilities.swift

```diff
@@ -8,6 +8,37 @@
 import Foundation
 
 extension URL {
+    /// Attempt to construct a URL from the passed string and validate its components
+    /// This will only allow http & https URLs with a full hostname
+    init?(validatedString string: String) {
+        guard let url = URL(string: string) else {
+            return nil
+        }
+        
+        var components = URLComponents()
+        components.scheme = url.scheme
+        components.host = url.host
+        components.path = url.path
```

Comment:
Question: is it not possible to check the URL's properties, rather than constructing a URLComponents?

Non-blocking.